app/models/email_address.rb

```diff
@@ -38,6 +38,18 @@ def find_with_email(email)
       find_by(email_fingerprint: email_fingerprint)
     end
 
+    def update_last_sign_in_at_on_user_id_and_email(user_id:, email:)
+      return nil if email.to_s.empty?
+
+      email = email.downcase.strip
+      email_fingerprint = create_fingerprint(email)
+      # rubocop:disable Rails/SkipsModelValidations
+      EmailAddress.where(user_id: user_id, email_fingerprint: email_fingerprint).update_all(
+        last_sign_in_at: Time.zone.now,
```

Comment:
Does this skip the normal AR callbacks? if so do we need to include `updated_at` ?

Reply:
Yes, good catch. Added in 16ca895.